### ISSUE TITLE
RSS feed contains too much data

### ISSUE THREAD

**Issue by davorg** [manwar/theweeklychallenge#179]
Most RSS feeds contain about 10 items.

    $ GET https://theweeklychallenge.org/rss.xml | grep '<item>' | wc -l
    1199

It's pulling in information from places that aren't useful.

* Go to https://perl.theplanetarium.org/
* Turn off the "all feeds" checkbox
* Turn on the "Weekly Challenge" checkbox

Currently, I'm seeing items for obscure programming languages. I think they come from your `/tags/` directory. There must be a way to limit RSS items to just the `/blog/` directory.

**Comment by rjt-pl** [comment]
It looks like rss.xml contains _every_ post we've ever published, so I took a look at config.toml. We have rssLimit set to 10, but rssLimit was (mostly) removed in hugo 0.55.

There is more information about it, with some recommended fixes from the developer, here: https://discourse.gohugo.io/t/rsslimit-not-working-in-hugo-0-55/18096/5

Hope this helps.

**Comment by davorg** [comment]
So, yeah, I think there are two different issues going on here:

1. Hugo has changed the way that the size of an RSS file is limited. That can probably be fixed simply enough by following suggestions from @rjt-pl above
2. The feed contains items from directories that aren't really of interest. I can't see why stuff from `/tags/` would ever be useful in the feed. I think you probably want to limit it to things in `/blog/` (but I don't know the layout of the site that well, so there may be other useful directories)

**Comment by manwar** [comment]
@davorg @rjt-pl I have tweaked the script the generates rss document. We have now document with 1071 items.
Please review if this is acceptable.

**Comment by davorg** [comment]
There's really no standard here. Just as long as it's a few tens of items and not hundreds. 

The usual approach is to get a list of the most recent items, sort it by date and truncate it to 10 (or 20) items.

**Comment by rjt-pl** [comment]
I agree with @davorg -- sort by date, grab the most recent 10-20 items only, which would be about 2 months' worth of posts based on the 2x/week average. The link I provided looks like it should be able to help you re-instate the rssLimit accordingly.

I'd love to get hands-on myself, but I'm traveling at the moment. :-)

**Comment by manwar** [comment]
@davorg @rjt-pl We now have 24 latest entries.

**Comment by davorg** [comment]
It looks like you have manually updated the RSS file to remove most of it.

https://github.com/manwar/theweeklychallenge/commit/b68540ef8e634ec5566fb23c00e580f9e1acdfc1

But isn't that automatically generated? Won't we get the full version back, the next time the generation code runs?

**Comment by manwar** [comment]
@davorg `RSS` feed for the site was never auto generated. I have a tiny `Perl` script (private) that generates the feed using `XML::Feed` module.

At first, I was pulling everything under the `/docs` folder and that was generating roughly `1199` items as you pointed out earlier. Then I trimmed it down to only pull things from `/docs/blog` folder. This then brought the count down to `1071` items. Still a very high numbers as you correctly suggested.

I then decided to pull only last `2 months` items from the `/docs/blog` folder. This worked out nicely and we got `24` items. However I noticed it pulled some really old items and not just `2 months old`. I did further investigation and found out that items under the `/docs/blog` folder items timestamp changes every time I rebuild the site. So this is unreliable. Then I turned to the source folder `/content/blog`. Here the timestamp is more reliable than the `/docs/blog` folder. I then came up with logic to map the items in both folder and use the timestamp which is more reliable i.e. `/content/blog` folder.

Now when I generate the feed it has the right amount of items and only the last `2 months` items. Usually I update the feed as soon as I release new weekly challenge i.e. `Monday`  morning mostly.

**Comment by davorg** [comment]
Aha. That all makes sense. Thanks. It was the absence of the RSS-building program that was confusing me. Would you consider adding the code to the repo?

Thanks for working on this.

> items under the `/docs/blog` folder items timestamp changes every time I rebuild the site

That sounds like a bad idea :-) Is this a Hugo "feature" or is it code of yours that we could look at fixing?

**Comment by manwar** [comment]
@davorg The timestamp related issue is likely `Hugo` feature as I am not doing anything related to it myself.
As requested, I have now added the script to the repository.

https://github.com/manwar/theweeklychallenge/blob/master/script/generate-rss-feed

**Comment by davorg** [comment]
Having looked at this a bit more, I think relying on the filesystem timestamp of the files in the `/content/blog` directory is extremely fragile. I checked out the repo yesterday and all of my files have a timestamp from yesterday. So if you were ever to get a new PC and a new local checkout of the repo, you would lose all that information - which seems scary to me!

But. You have the timestamp in a `date` field in the frontmatter of the files. So I think that's a much safer bet. I'll have a little prod at using that in RSS generation program.

**Comment by davorg** [comment]
I think we improved this massively when the RSS generation program was changed to only include files from the `/docs/blog` directory. There's still a potential problem about the dates on blog entries, but I'll close this issue and raise a separate one for that.
